test/e2e/e2e_test.go

```diff
@@ -335,26 +316,63 @@ func verifyResources(t *testing.T, awaitility *wait.Awaitility, murName string,
 		assert.Equal(t, userAccount.Name, rb.Subjects[0].Name)
 		assert.Equal(t, "edit", rb.RoleRef.Name)
 	}
+
+	// Get member cluster to verify that it was used to provision user accounts
+	memberCluster, ok, err := hostAwait.GetKubeFedCluster(cluster.Member, nil)
+	require.NoError(t, err)
+	require.True(t, ok)
+
+	// Then finally check again the MasterUserRecord with the expected (embedded) UserAccount status, on top of the other criteria
+	expectedEmbededUaStatus := toolchainv1alpha1.UserAccountStatusEmbedded{
+		Cluster: toolchainv1alpha1.Cluster{
+			Name:        mur.Spec.UserAccounts[0].TargetCluster,
+			APIEndpoint: memberCluster.Spec.APIEndpoint,
+			ConsoleURL:  expectedConsoleURL(t, memberAwait, memberCluster),
+		},
+		UserAccountStatus: userAccount.Status,
+	}
+	_, err = hostAwait.WaitForMasterUserRecord(mur.Name,
+		wait.UntilMasterUserRecordHasConditions(provisioned()),
+		wait.UntilMasterUserRecordHasUserAccountStatuses(expectedEmbededUaStatus))
+	assert.NoError(t, err)
 }
 
 func toIdentityName(userID string) string {
 	return fmt.Sprintf("%s:%s", "rhd", userID)
 }
 
-func registration(t *testing.T, route string, identity authsupport.Identity) {
-	// Call signup endpoint with an valid token.
-	emailClaim := authsupport.WithEmailClaim(uuid.NewV4().String() + "@email.tld")
-	iatClaim := authsupport.WithIATClaim(time.Now().Add(-60 * time.Second))
-	token, err := authsupport.GenerateSignedE2ETestToken(identity, emailClaim, iatClaim)
-	require.NoError(t, err)
-
-	req, err := http.NewRequest("POST", route+"/api/v1/signup", nil)
-	require.NoError(t, err)
-	req.Header.Set("Authorization", "Bearer "+token)
-	req.Header.Set("content-type", "application/json")
-	client := getClient()
+func expectedConsoleURL(t *testing.T, memberAwait *wait.MemberAwaitility, cluster v1beta1.KubeFedCluster) string {
+	// If OpenShift 3.x console available then we expect its URL in the status
+	consoleURL := openShift3XConsoleURL(cluster.Spec.APIEndpoint)
```

Comment:
what about moving all this logic of OpenShift 3.x vs OpenShift 4.x console URL in the `MemberAwaitility.GetConsoleRoute()` ?

Reply:
I thought that’s it would be easier to remove the 3.x functionality later (when we don’t need it anymore) if it’s separated from the rest of the code.